roles/mining-proxy/src/lib/downstream_mining.rs

```diff
@@ -487,40 +487,49 @@ use network_helpers_sv2::plain_connection_tokio::PlainConnection;
-        let (receiver, sender): (Receiver<EitherFrame>, Sender<EitherFrame>) =
-            PlainConnection::new(stream).await;
-        let node = DownstreamMiningNode::new(receiver, sender, ids.next());
-
-        task::spawn(async move {
-            let mut incoming: StdFrame = node.receiver.recv().await.unwrap().try_into().unwrap();
-            let message_type = incoming.get_header().unwrap().msg_type();
-            let payload = incoming.payload();
-            let routing_logic = super::get_common_routing_logic();
-            let node = Arc::new(Mutex::new(node));
-
-            // Call handle_setup_connection or fail
-            match DownstreamMiningNode::handle_message_common(
-                node.clone(),
-                message_type,
-                payload,
-                routing_logic,
-            ) {
-                Ok(SendToCommon::RelayNewMessageToRemote(_, message)) => {
-                    let message = match message {
-                        roles_logic_sv2::parsers::CommonMessages::SetupConnectionSuccess(m) => m,
+    let mut should_continue = true;
+    while should_continue {
+        tokio::select! {
+            Ok((stream,_)) = listener.accept() => {
+                let (receiver, sender): (Receiver<EitherFrame>, Sender<EitherFrame>) =
+                    PlainConnection::new(stream).await;
+                let node = DownstreamMiningNode::new(receiver, sender, ids.next());
+
+                task::spawn(async move {
+                    let mut incoming: StdFrame = node.receiver.recv().await.unwrap().try_into().unwrap();
+                    let message_type = incoming.get_header().unwrap().msg_type();
+                    let payload = incoming.payload();
+                    let routing_logic = super::get_common_routing_logic();
+                    let node = Arc::new(Mutex::new(node));
+
+                    // Call handle_setup_connection or fail
+                    match DownstreamMiningNode::handle_message_common(
+                        node.clone(),
+                        message_type,
+                        payload,
+                        routing_logic,
+                    ) {
+                        Ok(SendToCommon::RelayNewMessageToRemote(_, message)) => {
+                            let message = match message {
+                                roles_logic_sv2::parsers::CommonMessages::SetupConnectionSuccess(m) => m,
+                                _ => panic!(),
+                            };
+                            DownstreamMiningNode::start(node, message).await
+                        }
                         _ => panic!(),
-                    };
-                    DownstreamMiningNode::start(node, message).await
-                }
-                _ => panic!(),
+                    }
+                });
+            },
+            _ = &mut shutdown_rx => {
+                info!("Closing listener on {}", address);
```

Comment:
s/info!/error!

Reply:
This is the expected behavior, to stop and return from that function, leading to clean exit.